File: backend/Models/Asset.cs
namespace SteamLibAssets.Models
{
    public class Asset
    {
        public string FullImageUrl { get; set; } = "";
        public string ThumbnailImageUrl { get; set; } = "";
        public int Width { get; set; }
        public int Height { get; set; }
        public Author Author { get; set; } = new Author { Name = "", AvatarUrl = "", SteamProfileUrl = "" };
    }
}

File: frontend/src/types/Asset.ts
export interface Asset {
  fullImageUrl: string;
  thumbnailImageUrl: string;
  width: number;
  height: number;
  author: {
    name: string;
    avatarUrl: string;
    steamProfileUrl: string;
  };
}

export type AssetType = "grid" | "icon" | "hero" | "logo";

export type PaginatedAssets = {
  page: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
  assets: Asset[];
};

export type DownloadAssets = {
  grid: Asset;
  hero: Asset;
  logo: Asset;
  icon: Asset;
};
